model_zoo/app_integrate_bundle/ensemble.py

```diff
@@ -119,54 +110,54 @@ def ensemble_inference(self, device, test_datalist, ensemble="Mean"):
         logger.info(f"Total Records in Test Dataset: {len(test_datalist)}")
 
         bundle_inference_config_path = os.path.join(self.bundle_path, "configs", inference_config_paths[0])
-        bundle_inference_config = ConfigParser()
-        bundle_inference_config.read_config(bundle_inference_config_path)
-        bundle_inference_config.update({Const.KEY_BUNDLE_ROOT: self.bundle_path})
-        bundle_inference_config.update({Const.KEY_INFERENCE_DATASET_DATA: test_datalist})
+        inference_workflow = ConfigWorkflow(
+            config_file=bundle_inference_config_path,
+            meta_file=None,
+            logging_file=None,
+            workflow="inference",
+        )
+        inference_workflow.dataset_data = test_datalist
+        # this application has an additional requirement for the bundle workflow to provide the property dataloader
+        inference_workflow.add_property(name="dataloader", required=True, config_id="dataloader")
+        inference_workflow.initialize()
 
         # update postprocessing with mean ensemble or vote ensemble
-        post_transform = bundle_inference_config.config["postprocessing"]
-        ensemble_transform = {
-            "_target_": f"{ensemble}Ensembled",
-            "keys": ["pred", "pred", "pred", "pred", "pred"],
-            "output_key": "pred",
-        }
+        _ensemble_transform = MeanEnsembled if ensemble == "Mean" else VoteEnsembled
+        ensemble_transform = _ensemble_transform(keys=["pred"] * args.n_splits, output_key="pred")
         if ensemble == "Mean":
-            post_transform["transforms"].insert(0, ensemble_transform)
+            _postprocessing = Compose((ensemble_transform, inference_workflow.postprocessing))
         elif ensemble == "Vote":
-            post_transform["transforms"].insert(-1, ensemble_transform)
+            _postprocessing = Compose((inference_workflow.postprocessing, ensemble_transform))
         else:
             raise NotImplementedError
-        bundle_inference_config.update({Const.KEY_INFERENCE_POSTPROCESSING: post_transform})
 
         # update network weights
-        _networks = [bundle_inference_config.get_parsed_content("network")] * 5
         networks = []
-        for i, _network in enumerate(_networks):
-            _network.load_state_dict(torch.load(self.bundle_path + f"/models/model{i}.pt"))
+        for i in range(args.n_splits):
+            _network = inference_workflow.network_def.to(device)
+            _network.load_state_dict(torch.load(self.bundle_path + f"/models/model_fold{i}.pt", map_location=device))
             networks.append(_network)
 
         evaluator = EnsembleEvaluator(
```

Comment:
I would suggest to:
```py
inference_workflow.evaluator = EnsembleEvaluator(...)
inference_workflow.run()
inference_workflow.finalize()
```
What do you think?

Thanks.

Reply:
LGTM, have updated it in the latest commit. Thanks!